pkg/imageregistry/imageregistry.go

```diff
@@ -0,0 +1,233 @@
+			ObjectMeta: metav1.ObjectMeta{
+				Name: imageRegistryObjName,
+			},
+		},
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("imageRegistry object %s doesn't exist", imageRegistryObjName)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Get fetches existing imageRegistry from cluster.
+func (builder *Builder) Get() (*imageregistryv1.Config, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Pulling existing imageRegistry with name %s from cluster", builder.Definition.Name)
+
+	imageRegistry := &imageregistryv1.Config{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+		Name: builder.Definition.Name,
+	}, imageRegistry)
+
+	if err != nil {
+		glog.V(100).Infof("ImageRegistry object %s doesn't exist", builder.Definition.Name)
+
+		return nil, err
+	}
+
+	return imageRegistry, nil
+}
+
+// Exists checks whether the given imageRegistry exists.
+func (builder *Builder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if imageRegistry %s exists", builder.Definition.Name)
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Update renovates a imageRegistry in the cluster and stores the created object in struct.
+func (builder *Builder) Update() (*Builder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Updating the imageRegistry %s", builder.Definition.Name)
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("imageRegistry object %s doesn't exist", builder.Definition.Name)
+	}
+
+	err := builder.apiClient.Update(context.TODO(), builder.Definition)
+	if err == nil {
+		builder.Object = builder.Definition
+	}
+
+	return builder, err
+}
+
+// GetManagementState fetches imageRegistry ManagementState.
+func (builder *Builder) GetManagementState() (v1.ManagementState, error) {
+	if valid, err := builder.validate(); !valid {
+		return "", err
+	}
+
+	glog.V(100).Infof("Getting imageRegistry ManagementState configuration")
+
+	if !builder.Exists() {
+		return "", fmt.Errorf("imageRegistry object doesn't exist")
+	}
+
+	return builder.Object.Spec.ManagementState, nil
+}
+
+// WithManagementState returns true when succeeded to change imageRegistry operator management state.
+func (builder *Builder) WithManagementState(expectedManagementState v1.ManagementState) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Setting imageRegistry %s with expectedManagementState: %v",
+		builder.Definition.Name, expectedManagementState)
+
+	builder.Definition.Spec.ManagementState = expectedManagementState
+
+	return builder
+}
+
+// SetManagementState returns true when succeeded to change imageRegistry operator management state.
+func (builder *Builder) SetManagementState(expectedManagementState v1.ManagementState) error {
```

Comment:
This function name and log entries make it similar and confusing with the above function (WithManagementState)
```suggestion
// ReviewSetManagementState adjusts the management state and verified the new state is set.
func (builder *Builder) ReviewSetManagementState(expectedManagementState v1.ManagementState) error {
```

Reply:
why?

Reply:
anyway it kind of helper for setting managementState with the verification it was really set.
if you don't like it like this I can move it to the eco-gotests as well

Reply:
originally WithManagementState method was not planed. This method had to includes the whole procedure.
But, according to the goinfra architecture we are using With... scheme, so I changed it

Reply:
moving this method to the eco-gotests repo